services/motion/client.go

```diff
@@ -97,6 +101,66 @@ func (c *client) MoveOnMap(
 	return resp.Success, nil
 }
 
+func (c *client) MoveOnGlobe(
+	ctx context.Context,
+	componentName resource.Name,
+	destination *geo.Point,
+	heading float64,
+	movementSensorName resource.Name,
+	obstacles []*spatialmath.GeoObstacle,
+	linearVelocity float64,
+	angularVelocity float64,
+	extra map[string]interface{},
+) (bool, error) {
+	ext, err := vprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return false, err
+	}
+
+	if destination == nil {
+		return false, errors.New("Must provide a destination")
+	}
+
+	req := &pb.MoveOnGlobeRequest{
+		Name:               c.name,
+		ComponentName:      protoutils.ResourceNameToProto(componentName),
+		Destination:        &commonpb.GeoPoint{Latitude: destination.Lat(), Longitude: destination.Lng()},
+		MovementSensorName: protoutils.ResourceNameToProto(movementSensorName),
+		Extra:              ext,
+	}
+
+	// Optionals
+	if !math.IsNaN(heading) {
+		req.Heading = &heading
+	}
+	if len(obstacles) > 0 {
```

Comment:
I don't think the if statement is necessary here

Reply:
It's mildly more efficient to have it in